File: app/src/main/java/model/Notification.java
package model;

import androidx.annotation.Keep;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import java.io.Serializable;

import ir.hamsaa.persiandatepicker.api.PersianPickerDate;
import ir.hamsaa.persiandatepicker.date.PersianDateImpl;
import utils.DateHelper;

@Keep
@Entity(tableName = "notifications")
public class Notification implements Serializable {

    @ColumnInfo(name = "id")
    @PrimaryKey()
    private int id;

    @ColumnInfo(name = "content")
    private String content;

    @ColumnInfo(name = "is_done")
    private boolean isDone;

    @ColumnInfo(name = "is_shown", defaultValue = "false")
    private boolean isShown;

    @ColumnInfo(name = "arrive_date", defaultValue = "0")
    private long arriveDate;

    @ColumnInfo(name = "priority")
    private Priority priority;

    @ColumnInfo(name = "category")
    private String category;

    @ColumnInfo(name = "category_id", defaultValue = "0")
    private int categoryId;

    @Ignore
    private DateTime dateTime;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public boolean isDone() {
        return isDone;
    }

    public void setDone(boolean done) {
        isDone = done;
    }

    public boolean isShown() {
        return isShown;
    }

    public void setShown(boolean shown) {
        isShown = shown;
    }

    public long getArriveDate() {
        return arriveDate;
    }

    public void setArriveDate(long arriveDate) {
        this.arriveDate = arriveDate;
    }

    public Priority getPriority() {
        return priority;
    }

    public void setPriority(Priority priority) {
        this.priority = priority;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public int getCategoryId() {
        return categoryId;
    }

    public void setCategoryId(int categoryId) {
        this.categoryId = categoryId;
    }

    public DateTime getDateTime() {
        dateTime = new DateTime();

        if (arriveDate == 0)
            return dateTime;

        PersianPickerDate persianDate = new PersianDateImpl();
        persianDate.setDate(arriveDate);

        dateTime.setDate(persianDate);

        DateHelper dateHelper = new DateHelper(arriveDate);

        dateTime.setHour(dateHelper.getHour());
        dateTime.setMinute(dateHelper.getMinute());

        return dateTime;
    }

    public void setDateTime(DateTime dateTime) {
        this.dateTime = dateTime;
    }

    public String getClock() {
        DateHelper dateHelper = new DateHelper(arriveDate);
        return dateHelper.getClock();
    }

}

File: app/src/main/java/model/Notification.kt
package model

import androidx.annotation.Keep
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.io.Serializable

@Keep
@Entity(tableName = "notifications")
data class Notification(
    @ColumnInfo(name = "id")
    @PrimaryKey
    var id: Int = 0,

    @ColumnInfo(name = "content")
    var content: String? = null,

    @ColumnInfo(name = "is_done")
    var isDone: Boolean = false,

    @ColumnInfo(name = "is_shown", defaultValue = "false")
    var isShown: Boolean = false,

    @ColumnInfo(name = "arrive_date", defaultValue = "0")
    var arriveDate: Long = 0,

    @ColumnInfo(name = "priority")
    var priority: Priority? = null,

    @ColumnInfo(name = "category")
    var category: String? = null,

    @ColumnInfo(name = "category_id", defaultValue = "0")
    var categoryId: Int = 0
) : Serializable {

    fun initWith(initTodo: Todo?) {
        if (initTodo == null)
            return

        id = initTodo.id
        content = initTodo.title
        isDone = initTodo.isDone
        isShown = false
        arriveDate = initTodo.arriveDate
        priority = initTodo.priority
        category = initTodo.category
        categoryId = initTodo.categoryId
    }
}